cli/internal/renderer/graph/tree/renderer.go

```diff
@@ -0,0 +1,124 @@
+package tree
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/jedib0t/go-pretty/v6/list"
+
+	syncdag "ocm.software/open-component-model/bindings/go/dag/sync"
+	"ocm.software/open-component-model/cli/internal/renderer/graph"
+)
+
+type RendererOptions[T cmp.Ordered] struct {
```

Comment:
nit: please move options to separate `renderer_options.go` and add documentation on when and how to use them